src/main/python/ayab/plugins/ayab_plugin/ayab_control.py

```diff
@@ -370,82 +376,95 @@ def __middlecoltwice(self, lineNumber, imgHeight, lenImgExpanded):
 
         # Double the line minus the 2 you save on the begin
         # and end of each imgRow
-        passesPerRow = 2 * self.__numColors - 2
+        passesPerRow = 2 * self._numColors - 2
+
+        q, r = divmod(lineNumber, passesPerRow)
+        firstCol = (r == 0)
+        lastCol = (r == passesPerRow - 1)
 
-        imgRow = self.__startLine + lineNumber // passesPerRow
+        imgRow = self._startLine + q
 
-        if self.__infRepeat:
+        if self._infRepeat:
             imgRow = imgRow % imgHeight
 
-        if imgRow % 2 != 0:
-            color = ((lineNumber % passesPerRow) + 1) // 2
+        if firstCol or lastCol:
+            color = (lastCol + q) % 2
         else:
-            color = (passesPerRow - (lineNumber % passesPerRow)) // 2
+            color = (r + 3) // 2
 
-        indexToSend = imgRow * self.__numColors + color
+        indexToSend = imgRow * self._numColors + color
 
-        sendBlankLine = not (lineNumber % passesPerRow == 0) and not ((lineNumber + 1) % passesPerRow == 0) and not (lineNumber % 2 == 0)
+        sendBlankLine = not firstCol and not lastCol and odd(lineNumber)
 
-        lastLine = (imgRow == imgHeight - 1) and (lineNumber % passesPerRow == passesPerRow - 1)
+        lastLine = (imgRow == imgHeight - 1) and lastCol
 
-        return color, imgRow, indexToSend, sendBlankLine, lastLine
+        return imgRow, color, indexToSend, sendBlankLine, lastLine
 
     # doublebed, multicolor <3 of pluto
-    # advances imgRow as soon as possible
-    def __heartofpluto(self, lineNumber, imgHeight, lenImgExpanded):
+    # rotates middle colors
+    def _heartofpluto(self, lineNumber, imgHeight, lenImgExpanded):
 
-        # Double the line minus the 2 you save from
-        # early advancing to next row
-        passesPerRow = 2 * self.__numColors - 2
+        # doublebed <3 of pluto multicolor
+        # 0-00 1-11 2-22 3-33 4-44 5-55 .. (imgRow)
+        # 0123 4567 8911 1111 1111 2222.. (lineNumber)
+        #             01 2345 6789 0123
+        #
```

Comment:
I added this multiline comment to help show what I believe the Heart of Pluto mode should be doing. Is it correct?